packages/udt/src/udt/index.ts

```diff
@@ -274,33 +367,514 @@ export class Udt extends ssri.Trait {
     return resTx;
   }
 
-  async completeChangeToLock(
+  /**
+   * Checks if a cell is a valid UDT cell for this token.
+   * A valid UDT cell must have this UDT's type script and contain at least 16 bytes of output data
+   * (the minimum required for storing the UDT balance as a 128-bit little-endian integer).
+   *
+   * @param cellOutputLike - The cell output to check
+   * @param outputData - The output data of the cell
+   * @returns True if the cell is a valid UDT cell for this token, false otherwise
+   *
+   * @example
+   * ```typescript
+   * const udt = new Udt(codeOutPoint, scriptConfig);
+   * const cellOutput = { lock: someLock, type: udt.script };
+   * const outputData = ccc.numLeToBytes(1000, 16); // 1000 UDT balance
+   *
+   * const isValid = udt.isUdtCell(cellOutput, outputData);
+   * console.log(`Is valid UDT cell: ${isValid}`); // true
+   * ```
+   *
+   * @remarks
+   * The method checks two conditions:
+   * 1. The cell's type script matches this UDT's script
+   * 2. The output data is at least 16 bytes long (required for UDT balance storage)
+   */
+  isUdtCell(cellOutputLike: ccc.CellOutputLike, outputData: ccc.HexLike) {
```

Comment:
Why not using [CellLike](https://api.ckbccc.com/types/_ckb_ccc_core.index.ccc.CellLike.html)?

Reply:
You make a fair point, it's not a given this entities have a Outpoint. In my implementation it did, cause it was only used on on-chain cells